### PR TITLE
feat: [FTL-7231] random username

### PR DESCRIPTION
**Pull request by anton-iskryzhytskyi**

### CODE REVIEW

**Review by yaroslava-kurash — APPROVED** [review]

File: common-libs/user-management/src/UserManagementService.ts [hunk 1]
```diff
@@ -177,17 +194,25 @@ export class UserManagementService {
 
   async completeSignUpForEmailOrPhone(token: string, confirmationCode: string) {
     const { login, shortPassword } = this.parseSignUpToken(token)
-    this._loginShouldBeEmailOrPhoneNumber(login)
```
Comment (Volland) [inline common-libs/user-management/src/UserManagementService.ts:180]: @anton-iskryzhytskyi why we remove this check?

File: common-libs/user-management/src/UserManagementService.ts [hunk 2]
```diff
@@ -177,17 +194,25 @@ export class UserManagementService {
...
 
-  async doesUnconfirmedUserExist(username: string) {
-    const normalizedUsername = normalizeUsername(username)
-    return this._cognitoIdentityService.doesUnconfirmedUserExist(normalizedUsername)
+  /**
+   * Uses an ep on KeysService to check if record exists or not for given username or alias (email / phone)
+   * @param login
+   */
+  async doesUnconfirmedUserExist(login: string) {
+    const { userExists, isUnconfirmed } = await this.doesUserExist(login)
```
Comment (Volland) [inline common-libs/user-management/src/UserManagementService.ts:207]: @anton-iskryzhytskyi, how we know that user is unconfirmed? As i understand, we query users by attributes but do we have a way to check the confirmation status?
We could check attributes or do a doesConfirmedUser exist call in pair with this method

Comment (anton-iskryzhytskyi) [reply common-libs/user-management/src/UserManagementService.ts:207]: to answer on your question I'd like to send you to the implementation of AWB there
`const isUnconfirmed = userExists && !(user.UserStatus === 'CONFIRMED')`

File: common-libs/user-management/src/UserManagementService.ts (hunk 1 above)
Comment (anton-iskryzhytskyi) [reply common-libs/user-management/src/UserManagementService.ts:180]: Haw it was:
signUp token had a format 'login::password', then we derive username from login, we need username to complete sign-up
Now:
signUp token has a format 'username::password', username is uuid, so can't be email or phone

File: common-libs/user-management/src/UserManagementService.ts (hunk 2 above)
Comment (Volland) [reply common-libs/user-management/src/UserManagementService.ts:207]: oki got it

**Review by Volland — APPROVED** [review]
looks good

**Pull request merged**